cmd/nsm-sriov-manager/nsm-sriov-manager.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	netDirectory    = "/sys/class/net/"
+	pciDirectory    = "/sys/bus/pci/devices"
+	sriovCapable    = "/sriov_totalvfs"
+	sriovConfigured = "/sriov_numvfs"
+	// nsmSRIOVDefaultNetworkServiceName defines a default network service name for all
+	// SRIOV VFs found on a host. Further editing might be required to map VF to real
+	// Network Services
+	nsmSRIOVDefaultNetworkServiceName = "networkservicemesh.io/sriov/nsm-default"
+)
+
+// VF describes a single instance of VF
+type VF struct {
+	NetworkService string `yaml:"networkService" json:"NetworkService"`
+	pciAddr        string // `yaml:"pciAddr" json:"pciAddr"`
+	ParentDevice   string `yaml:"parentDevice" json:"parentDevice"`
```

Comment:
True for SRIOV devices (they have PFs) but I can pass through an entire device, not just SRIOV VFs.  For that case I would assume this has a magic value (e.g. "")

Reply:
@edwarnicke made me aware of the case of a physical nic, I guess pci passthrough is the same case and it is not covered by this code, it is VF specific, I think physical NIC case should be addressed by a separate controller.